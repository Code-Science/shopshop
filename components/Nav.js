import React, { useState, useEffect } from 'react';
import classNames from 'classnames';
import Link from 'next/link';
import PropTypes from 'prop-types';
import styles from '../styles/components/Nav.module.css';

function Nav({ open }) {
  const [womenLinks, setWLinks] = useState(false);
  const [menLinks, setMLinks] = useState(false);

  const [startAnimation, setAnimation] = useState(false);

  const showOrHideLinks = (event, linksFor) => {
    if (window && window.innerWidth > 900) {
      if (linksFor === 'women') {
        setWLinks((prevState) => !prevState);
        setMLinks(false);
      } else {
        setWLinks(false);
        setMLinks((prevState) => !prevState);
      }
      return;
    }
    if (linksFor === 'women') {
      setWLinks((prevState) => !prevState);
    } else {
      setMLinks((prevState) => !prevState);
    }
  };

  useEffect(() => {
    if (open) setAnimation(true);
  }, [open]);

  const navPopupClassW = classNames({
    [styles.navPopup]: true,
    [styles.open]: womenLinks,
  });

  const navPopupClassM = classNames({
    [styles.navPopup]: true,
    [styles.open]: menLinks,
  });

  const nav = classNames({
    [styles.nav]: true,
    [styles.openNav]: open,
    [styles.closeNav]: !open && startAnimation,
  });

  return (
    <div className={nav}>
      <ul className={styles.navList}>
        <li className={styles.navItem}>
          <button
            type="button"
            className={styles.navName}
            onClick={(event) => showOrHideLinks(event, 'women')}
          >
            WOMEN
          </button>
          <span className={navPopupClassW}>
            <Link href="/collection/women/clothes">
              <a className={styles.popupItems}>Clothes</a>
            </Link>
            <Link href="/collection/women/accessories">
              <a className={styles.popupItems}>Accessories</a>
            </Link>
          </span>
        </li>
        <li className={styles.navItem}>
          <button
            type="button"
            className={styles.navName}
            onClick={(event) => showOrHideLinks(event, 'men')}
          >
            MEN
          </button>
          <span className={navPopupClassM}>
            <Link href="/collection/men/clothes">
              <a className={styles.popupItems}>Clothes</a>
            </Link>
            <Link href="/collection/men/accessories">
              <a className={styles.popupItems}>Accessories</a>
            </Link>
          </span>
        </li>
        <li className={styles.navItem}>
          <Link href="/collection/kids/clothes">
            <a className={styles.items}>KIDS</a>
          </Link>
        </li>
        <li className={styles.navItem}>
          <Link href="/">
            <a className={styles.items}>CONTACT</a>
          </Link>
        </li>
      </ul>
    </div>
  );
}

Nav.propTypes = {
  open: PropTypes.bool.isRequired,
};

export default Nav;
