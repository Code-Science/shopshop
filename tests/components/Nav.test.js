import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import Nav from '../../components/Nav';

jest.mock('../../store/StateProvider');

configure({ adapter: new Adapter() });

describe('Nav Component', () => {
  const wrapper = shallow(<Nav open={true} />);

  it('expect to render visible (desktop) Nav component', () => {
    expect(wrapper.length).toEqual(1);
  });
  it('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render hidden (mobile) Nav component', () => {
    expect(shallow(<Nav open={false} />).length).toEqual(1);
  });
});
