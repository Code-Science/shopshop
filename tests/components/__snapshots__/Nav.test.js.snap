// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component snapshot test 1`] = `
<div
  className="nav openNav"
>
  <ul
    className="navList"
  >
    <li
      className="navItem"
    >
      <button
        className="navName"
        onClick={[Function]}
        type="button"
      >
        WOMEN
      </button>
      <span
        className="navPopup"
      >
        <Link
          href="/collection/women/clothes"
        >
          <a
            className="popupItems"
          >
            Clothes
          </a>
        </Link>
        <Link
          href="/collection/women/accessories"
        >
          <a
            className="popupItems"
          >
            Accessories
          </a>
        </Link>
      </span>
    </li>
    <li
      className="navItem"
    >
      <button
        className="navName"
        onClick={[Function]}
        type="button"
      >
        MEN
      </button>
      <span
        className="navPopup"
      >
        <Link
          href="/collection/men/clothes"
        >
          <a
            className="popupItems"
          >
            Clothes
          </a>
        </Link>
        <Link
          href="/collection/men/accessories"
        >
          <a
            className="popupItems"
          >
            Accessories
          </a>
        </Link>
      </span>
    </li>
    <li
      className="navItem"
    >
      <Link
        href="/collection/kids/clothes"
      >
        <a
          className="items"
        >
          KIDS
        </a>
      </Link>
    </li>
    <li
      className="navItem"
    >
      <Link
        href="/"
      >
        <a
          className="items"
        >
          CONTACT
        </a>
      </Link>
    </li>
  </ul>
</div>
`;
